service/dap/server.go

```diff
@@ -1673,13 +1830,16 @@ func (s *Server) resetHandlesForStoppedEvent() {
 
 // doCommand runs a debugger command until it stops on
 // termination, error, breakpoint, etc, when an appropriate
-// event needs to be sent to the client.
-func (s *Server) doCommand(command string) {
-	if s.debugger == nil {
-		return
-	}
+// event needs to be sent to the client. asyncSetupDone is
+// a channel that will be closed to signal that an
+// asynchornous command has completed setup or was interrupted
+// due to an error, so the server is ready to receive new requests.
+func (s *Server) doCommand(command string, asyncSetupDone chan struct{}) {
+	defer func() {
+		s.asyncCommandDone(asyncSetupDone)
+	}()
```

Comment:
defer s.asyncCommandDone(asyncSetupDone)

Wontt it be clear not to pass asyncSetupDone to asyncCommandDone given that it should be closed by s.debugger.Command in this case.

Reply:
Hmm, I am not sure anymore why I have it there. I think there was some intermediate version of this code, which had some error check and early return before Command was called. I think I lost nil debugger check in all the reshuffling. I should put it back, since it might be unsafe to rely on the editor not to issue certain commands when in noDebug mode or not past launch/attach.

Reply:
Sometimes debugger.Command doesn't close this channel.